Add files via upload

antonychiu2 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

antonychiu2 committed:
Add files via upload

Commit:
Auto commit for SQL_Injection-4937ad47-74a9-4949-836e-7bcbebab1b40

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/4937ad47-74a9-4949-836e-7bcbebab1b40)

Commit:
Auto commit for SQL_Injection-f04794a9-083f-4cac-a7cf-660a4e387a44

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/f04794a9-083f-4cac-a7cf-660a4e387a44)

Commit:
Auto commit for SQL_Injection-d7177a9f-4483-4ae7-96fb-b68dad201d95

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/d7177a9f-4483-4ae7-96fb-b68dad201d95)

Commit:
Auto commit for SQL_Injection-fbf80673-319b-4003-a709-3d04e6fbad68

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/fbf80673-319b-4003-a709-3d04e6fbad68)

Commit:
Auto commit for SQL_Injection-38096c2d-b4e9-4fa4-9826-2df752f959d2

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/38096c2d-b4e9-4fa4-9826-2df752f959d2)

Commit:
Auto commit for SQL_Injection-2fb571f0-cdb7-441d-9ea3-ea34c7a27911

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/2fb571f0-cdb7-441d-9ea3-ea34c7a27911)

Commit:
Auto commit for SQL_Injection-accc7b74-3f8b-4fde-9457-0fc8302d1515

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/accc7b74-3f8b-4fde-9457-0fc8302d1515)

Commit:
Auto commit for SQL_Injection-23af06d5-7248-44e2-a400-4ed81424885d

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/23af06d5-7248-44e2-a400-4ed81424885d)

Commit:
Auto commit for SQL_Injection-8226a6a7-eb94-4fc6-917b-bafb64497e0b

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/8226a6a7-eb94-4fc6-917b-bafb64497e0b)

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -38,6 +38,6 @@
         }, limitQuery).exec();
     },
     getGraphCountByZone: function (zone) {
-        return (graphDataModel.countDocuments({ 'zone': zone }).exec());
+        return (graphDataModel.countDocuments({ 'zone': mongoSanitize.sanitize({ data: zone }).data }).exec());
     },
 };
 
```

```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/8226a6a7-eb94-4fc6-917b-bafb64497e0b)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_links.js b/web_server/config/models/graphs_links.js
--- a/web_server/config/models/graphs_links.js
+++ b/web_server/config/models/graphs_links.js
@@ -34,7 +34,7 @@
     getGraphLinksByZone: function (zone) {
         let limitQuery = { 'links': 1 };
         return graphLinksModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
 };
 
```

```diff
diff --git a/web_server/config/models/graphs_links.js b/web_server/config/models/graphs_links.js
--- a/web_server/config/models/graphs_links.js
+++ b/web_server/config/models/graphs_links.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/accc7b74-3f8b-4fde-9457-0fc8302d1515)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -52,7 +52,7 @@
         let AWSregex = new RegExp('.*' + search + '$');
         if (listOnly) {
             promise = tpdModel.find({
-                'tld': { '$regex': AWSregex },
+                'tld': mongoSanitize.sanitize({ data: { '$regex': AWSregex } }).data,
             }, { 'tld': 1, 'zones.zone': 1 }).exec();
         } else {
             promise = tpdModel.find({ 'tld': { '$regex': AWSregex } }).exec();
 
```

```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/fbf80673-319b-4003-a709-3d04e6fbad68)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_docs.js b/web_server/config/models/graphs_docs.js
--- a/web_server/config/models/graphs_docs.js
+++ b/web_server/config/models/graphs_docs.js
@@ -31,7 +31,7 @@
     getGraphDocsByZone: function (zone) {
         let limitQuery = { 'docs': 1 };
         return graphDocsModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
 };
 
```

```diff
diff --git a/web_server/config/models/graphs_docs.js b/web_server/config/models/graphs_docs.js
--- a/web_server/config/models/graphs_docs.js
+++ b/web_server/config/models/graphs_docs.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/38096c2d-b4e9-4fa4-9826-2df752f959d2)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -36,7 +36,7 @@
     getTPDsByZone: function (zone, listOnly) {
         let promise;
         if (!listOnly) {
-            promise = tpdModel.find({ 'zones.zone': zone }).exec();
+            promise = tpdModel.find({ 'zones.zone': mongoSanitize.sanitize({ data: zone }).data }).exec();
         } else {
             promise = tpdModel.find({
                 'zones.zone': zone,
 
```

```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/23af06d5-7248-44e2-a400-4ed81424885d)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -34,7 +34,7 @@
     getGraphDataByZone: function (zone) {
         let limitQuery = { 'data': 1, 'errs': 1 };
         return graphDataModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
     getGraphCountByZone: function (zone) {
 
```

```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/4937ad47-74a9-4949-836e-7bcbebab1b40)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs.js b/web_server/config/models/graphs.js
--- a/web_server/config/models/graphs.js
+++ b/web_server/config/models/graphs.js
@@ -34,7 +34,7 @@
     getGraphConfigByZone: function (zone) {
         let limitQuery = { 'config': 1 };
         return graphModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
     getGraphCountByZone: function (zone) {
 
```

```diff
diff --git a/web_server/config/models/graphs.js b/web_server/config/models/graphs.js
--- a/web_server/config/models/graphs.js
+++ b/web_server/config/models/graphs.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/2fb571f0-cdb7-441d-9ea3-ea34c7a27911)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -45,7 +45,7 @@
         return (promise);
     },
     getTPDsByTPD: function (tpd) {
-        return tpdModel.findOne({ 'tld': tpd });
+        return tpdModel.findOne({ 'tld': mongoSanitize.sanitize({ data: tpd }).data });
     },
     getTPDsByWildcard: function (search, listOnly) {
         let promise;
 
```

```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/d7177a9f-4483-4ae7-96fb-b68dad201d95)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/cert_graphs.js b/web_server/config/models/cert_graphs.js
--- a/web_server/config/models/cert_graphs.js
+++ b/web_server/config/models/cert_graphs.js
@@ -31,6 +31,6 @@
 module.exports = {
     CertGraphModel: certGraphModel,
     getGraphDataByZone: function (zone) {
-        return certGraphModel.findOne({ 'zone': zone }).exec();
+        return certGraphModel.findOne({ 'zone': mongoSanitize.sanitize({ data: zone }).data }).exec();
     },
 };
 
```

```diff
diff --git a/web_server/config/models/cert_graphs.js b/web_server/config/models/cert_graphs.js
--- a/web_server/config/models/cert_graphs.js
+++ b/web_server/config/models/cert_graphs.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/932374a4-77bd-41fc-8dcf-7aefbdd74170/fix/f04794a9-083f-4cac-a7cf-660a4e387a44)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Commit:
Auto commit for SQL_Injection-6b152e77-1cd8-478c-80a5-c2c62c4b920b

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/c83906d3-dd68-4c54-8b00-13547fa795c4/fix/6b152e77-1cd8-478c-80a5-c2c62c4b920b)

Commit:
Auto commit for SQL_Injection-5dae1aee-a89b-4084-921e-d743d48d5e43

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/c83906d3-dd68-4c54-8b00-13547fa795c4/fix/5dae1aee-a89b-4084-921e-d743d48d5e43)

Commit:
Auto commit for SQL_Injection-9359f450-939a-44b2-8e73-87c96db4def8

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/c83906d3-dd68-4c54-8b00-13547fa795c4/fix/9359f450-939a-44b2-8e73-87c96db4def8)

Commit:
Auto commit for SQL_Injection-d48637c9-54f0-4e1a-9aca-4d13e7cb35e0

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/c83906d3-dd68-4c54-8b00-13547fa795c4/fix/d48637c9-54f0-4e1a-9aca-4d13e7cb35e0)

Commit:
Auto commit for SQL_Injection-0f89eda2-2008-4499-86e8-3310428c48e9

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/c83906d3-dd68-4c54-8b00-13547fa795c4/fix/0f89eda2-2008-4499-86e8-3310428c48e9)

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -40,7 +40,7 @@
             promise = tpdModel.find({ 'zones.zone': mongoSanitize.sanitize({ data: zone }).data }).exec();
         } else {
             promise = tpdModel.find({
-                'zones.zone': zone,
+                'zones.zone': mongoSanitize.sanitize({ data: zone }).data,
             }, { 'tld': 1, 'zones.zone': 1 }).exec();
         }
         return (promise);
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/c83906d3-dd68-4c54-8b00-13547fa795c4/fix/9359f450-939a-44b2-8e73-87c96db4def8)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -56,7 +56,7 @@
                 'tld': mongoSanitize.sanitize({ data: { '$regex': AWSregex } }).data,
             }, { 'tld': 1, 'zones.zone': 1 }).exec();
         } else {
-            promise = tpdModel.find({ 'tld': { '$regex': AWSregex } }).exec();
+            promise = tpdModel.find({ 'tld': mongoSanitize.sanitize({ data: { '$regex': AWSregex } }).data }).exec();
         }
         return (promise);
     },
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/c83906d3-dd68-4c54-8b00-13547fa795c4/fix/5dae1aee-a89b-4084-921e-d743d48d5e43)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_docs.js b/web_server/config/models/graphs_docs.js
--- a/web_server/config/models/graphs_docs.js
+++ b/web_server/config/models/graphs_docs.js
@@ -30,7 +30,7 @@
 module.exports = {
     GraphDocsModel: graphDocsModel,
     getGraphDocsByZone: function (zone) {
-        let limitQuery = { 'docs': 1 };
+        let limitQuery = { 'docs': mongoSanitize.sanitize({ data: 1 }).data };
         return graphDocsModel.findOne({
             'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/c83906d3-dd68-4c54-8b00-13547fa795c4/fix/d48637c9-54f0-4e1a-9aca-4d13e7cb35e0)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs.js b/web_server/config/models/graphs.js
--- a/web_server/config/models/graphs.js
+++ b/web_server/config/models/graphs.js
@@ -33,7 +33,7 @@
 module.exports = {
     GraphModel: graphModel,
     getGraphConfigByZone: function (zone) {
-        let limitQuery = { 'config': 1 };
+        let limitQuery = { 'config': mongoSanitize.sanitize({ data: 1 }).data };
         return graphModel.findOne({
             'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/c83906d3-dd68-4c54-8b00-13547fa795c4/fix/0f89eda2-2008-4499-86e8-3310428c48e9)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_links.js b/web_server/config/models/graphs_links.js
--- a/web_server/config/models/graphs_links.js
+++ b/web_server/config/models/graphs_links.js
@@ -33,7 +33,7 @@
 module.exports = {
     GraphLinksModel: graphLinksModel,
     getGraphLinksByZone: function (zone) {
-        let limitQuery = { 'links': 1 };
+        let limitQuery = { 'links': mongoSanitize.sanitize({ data: 1 }).data };
         return graphLinksModel.findOne({
             'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/c83906d3-dd68-4c54-8b00-13547fa795c4/fix/6b152e77-1cd8-478c-80a5-c2c62c4b920b)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.
